### PR TITLE
[NFTs] Track item's metadata depositor

### PR DESCRIPTION
**Pull request by jsidorenko**
As part of the upcoming feature to support the pre-signed claims those would allow to offload of the deposit onto the final NFT's receiver, we need to start tracking the metadata depositor in a similar way as we did for attributes.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2249855

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2249854

File: frame/nfts/src/features/attributes.rs [hunk 1]
```diff
@@ -159,6 +159,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				Collection::<T, I>::get(&collection).ok_or(Error::<T, I>::UnknownCollection)?;
 
 			if let Some(check_owner) = &maybe_check_owner {
+				// deposit.account.is_none(); maybe_check_owner.is_some()
```
Comment (joepetrowski) [inline frame/nfts/src/features/attributes.rs:162]: Why commented out code?

**Review by joepetrowski — APPROVED** [review]

File: frame/nfts/src/features/attributes.rs (hunk 1 above)
Comment (jsidorenko) [reply frame/nfts/src/features/attributes.rs:162]: It's not a code, just a note to understand the code below better :)

**Review by gilescope — APPROVED** [review]

File: frame/nfts/src/features/attributes.rs (hunk 1 above)
Comment (gilescope) [reply frame/nfts/src/features/attributes.rs:162]: ah that wasn't clear. I thought it was code for what was to come.

Comment (jsidorenko) [reply frame/nfts/src/features/attributes.rs]: changed

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2250833

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2250832

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2251058

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2251059

**Comment by jsidorenko** [comment]
bot merge

**Review by gilescope — APPROVED** [review]

**Comment by jsidorenko** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-38/2122/1

**Pull request merged**